crypto/asn1/a_d2i_fp.c

```diff
@@ -95,7 +95,7 @@ int asn1_d2i_read_bio(BIO *in, BUF_MEM **pb)
 {
     BUF_MEM *b;
     unsigned char *p;
-    int i;
+    ptrdiff_t i;
```

Comment:
ptrdiff_t is a C99 thing and we're still trying to use C89/C90. If you really want to use it we'll probably need to have support for that in the some header file for that platforms that don't provide it.

Reply:
BIO_read() also doesn't return a ptrdiff_t, so this doesn't even make sense to me.

Reply:
Yes, you're right. ptrdiff_t does not make sense here.